addons/purchase/views/purchase_views.xml

```diff
@@ -232,6 +232,8 @@
                                     <field name="sequence" widget="handle"/>
                                     <field
                                         name="product_id"
+                                        widget="product_label_section_and_note_field"
+                                        optional="show"
```

Comment:
Why make it `optional`? If anything, it's even weirder now to have the `name` optional, as it's included in the same column through the widget.

Reply:
You're absolutely right. Making product_id/product_tmpl_id `optional` doesn't make sense, especially since it's a required field. However, I believe making the `name` field optional could be beneficial, given that it's computed. Users has the flexibility to hide or show it, similar to what's been implemented in the invoicing!